File: tokilearning/Latihan/cekbilanganprima.cpp
#include <stdio.h>

using namespace std;

int main(){
    
    int n;
    bool prime;
    
    while (scanf("%d\n",&n)!=EOF){
          
          if (n<2 || n%2==0 && n!=2){
                   printf("TIDAK\n");
                   continue;
                   }
          prime = true;
          for (int i=3; i<n; i++)
              if (n%i==0){
                          prime = false;
                          break;
                          }
          if (prime) printf("YA\n");
             else printf("TIDAK\n");
          }
    return 0;
}
          

File: tokilearning/Latihan/cekbilanganprima.pas
var
  n,i:longint;
  prime:boolean;

begin
  while not eof do
    begin
      readln(n);
      if n<2 then
        begin
          writeln('TIDAK');
          continue;
        end;
      prime:=true;
      for i:=2 to trunc(sqrt(n)) do
        if n mod i=0 then
          begin
            prime:=false;
            break;
          end;
      if prime then writeln('YA')
        else writeln('TIDAK');
    end;
end.
